main.py

```diff
@@ -174,17 +183,37 @@ def broadcast_block(self, block_hash: str, block: Block):
             self.ez_send(peer, blockMessage)
 
 
+    def send_peers(self) -> None:
+        peers = self.get_peers()
+        peerMessage = PeersMessage(self.my_peer.mid)
+
+        for peer in peers:
+            self.ez_send(peer, peerMessage)
+
+
+    @lazy_wrapper(PeersMessage)
+    def receive_peers(self, peer: Peer, payload: PeersMessage) -> None:
+        if payload.mid not in self.known_peers_mid:
```

Comment:
Here since we are using set we can run 'add' without any problems and I think we should send all the peers in every case to others. So you can delete this if

Reply:
I think this have a double meaning, apart of no adding more than once the same mid, we are avoiding sending a mid we already sent